controller/src/main/java/io/pravega/controller/store/kvtable/InMemoryKVTMetadataStore.java

```diff
@@ -98,20 +94,26 @@ public Scope newScope(String scopeName) {
     @Override
     @Synchronized
     public CompletableFuture<Boolean> checkScopeExists(String scope) {
-        return CompletableFuture.completedFuture(scopes.containsKey(scope));
+        return Futures.completeOn(CompletableFuture.completedFuture(this.streamStore.getScopes().containsKey(scope)), executor);
     }
 
     @Override
     @Synchronized
     public CompletableFuture<Boolean> checkTableExists(String scopeName, String kvt) {
-        return CompletableFuture.completedFuture((InMemoryScope) getScope(scopeName)).thenApply(scope -> scope.checkTableExists(kvt));
+        return Futures.completeOn(checkScopeExists(scopeName).thenCompose(exists -> {
+            if (exists) {
+                return CompletableFuture.completedFuture((InMemoryScope) getScope(scopeName))
```

Comment:
you dont need to do `CompletableFuture.completedFuture().thenApply`.
Instead you can compute the value and then do CompletableFuture.completedFuture(value)

Reply:
sure.